src/Classes/Entities/EventEntity.php

```diff
@@ -13,20 +14,170 @@ class EventEntity
     protected $notes;
 
     public function __construct(
+        int $eventId = null,
         string $name = null,
         int $category = null,
         string $location = null,
-        Date $date = null,
-        Time $startTime = null,
-        Time $endTime = null,
+        string $date = null,
+        string $startTime = null,
+        string $endTime = null,
         string $notes = null
     ) {
+        $this->id = ($this->id ?? $eventId);
         $this->name = ($this->name ?? $name);
         $this->category = ($this->category ?? $category);
         $this->location = ($this->location ?? $location);
         $this->date = ($this->date ?? $date);
         $this->startTime = ($this->startTime ?? $startTime);
         $this->endTime = ($this->endTime ?? $endTime);
         $this->notes = ($this->notes ?? $notes);
+
+        $this->sanitiseData();
+    }
+
+    /**
+     * Will sanitise all the fields for an applicant.
+     */
+    private function sanitiseData()
+    {
+        $this->id = (int) $this->id;
+        $this->name = $this->sanitiseString($this->name);
+        $this->name = self::validateLength($this->name, 20);
+        $this->category = (int)$this->category;
+        $this->location = $this->sanitiseString($this->location);
+        $this->location = self::validateLength($this->location, 255);
+        $this->date = $this->sanitiseString($this->date);
+        $this->startTime = $this->sanitiseString($this->startTime);
+        $this->endTime = $this->sanitiseString($this->endTime);
+        $this->notes = $this->sanitiseString($this->notes);
```

Comment:
This should probably have a length check too.

Reply:
exist and length check maybe?

Reply:
Notes doesn't need to exist. Other fields do.